src/sql/src/plan/statement/ddl.rs

```diff
@@ -4751,8 +4766,24 @@ pub fn plan_alter_cluster(
                     if replica_defs.is_some() {
                         sql_bail!("REPLICAS not supported for managed clusters");
                     }
+                    if schedule.is_some()
+                        && !matches!(schedule, Some(ClusterScheduleOptionValue::Manual))
+                    {
+                        scx.require_feature_flag(&ENABLE_CLUSTER_SCHEDULE_REFRESH)?;
+                    }
 
                     if let Some(replication_factor) = replication_factor {
+                        if schedule.is_some()
+                            && !matches!(schedule, Some(ClusterScheduleOptionValue::Manual))
+                        {
+                            sql_bail!("REPLICATION FACTOR cannot be given together with any SCHEDULE other than MANUAL");
+                        }
+                        if let Some(current_schedule) = cluster.schedule() {
+                            if !matches!(current_schedule, ClusterScheduleOptionValue::Manual) {
+                                sql_bail!("REPLICATION FACTOR cannot be set if the cluster SCHEDULE is anything other than MANUAL");
+                            }
+                        }
```

Comment:
Can users create a cluster today with a non-manual schedule and specify the replication factor? If so, I think this is a backwards incompatible change and we might need to think carefully so we don't break an existing environment

Reply:
No, the `SCHEDULE = ...` syntax is not enabled in production for anybody, because no non-manual schedule was implemented before this PR.